### PR TITLE
docs: update PR template for adding new vendor checklist

### PR DESCRIPTION
**Pull request by darshit-s3**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] Backwards compatible
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation (If applicable)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
-- [ ] My code follows the style guidelines of this project
-- [ ] I have performed a self-review of my code
-- [ ] I have made corresponding changes to the documentation (If applicable)
+Please include a summary of the changes and the related issue to help is review the PR better and faster.
+
+# Checklist for adding new integration:
+
+- [ ] Defined `APIS` in constants [folder](../src/constants/instrumentation/).
+- [ ] Updated `SERVICE_PROVIDERS` in [common.ts](../src/constants/common.ts)
+- [ ] Created a folder under [instrumentation](../src/instrumentation/) with the name of the integration with atleast `patch.ts` and `index.ts` files.
```
Comment (rohit-kadhe) [inline .github/pull_request_template.md:9]: filename should be `instrumentation.ts`

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
+- [ ] Enabled instrumentation in [init.ts](../src/init/init.ts) & [types.ts](../src/init/types.ts) files.
```
Comment (rohit-kadhe) [inline .github/pull_request_template.md:10]: Reword to added instrumentation in `allInstrumentations` constant in `init.ts` and to the `InstrumentationType` in `types.ts`

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,20 +1,15 @@
...
+- [ ] Added examples for the new integration in the [examples](../src/examples/) folder.
+- [ ] Updated [package.json](../package.json) to install new dependencies for devDependencies.
+- [ ] Updated the [README.md](../README.md) of [langtrace-typescript-sdk](https://github.com/Scale3-Labs/langtrace-typescript-sdk) to include the new integration in the supported integrations table.
+- [ ] Updated the [README.md](https://github.com/Scale3-Labs/langtrace?tab=readme-ov-file#supported-integrations) of Langtrace's [repository](https://github.com/Scale3-Labs/langtrace) to include the new integration in the supported integrations table.
+- [ ] Added new integration page to supported integrations in [Langtrace Docs](https://github.com/Scale3-Labs/langtrace-docs)
```
Comment (rohit-kadhe) [inline .github/pull_request_template.md:15]: Any pr's that are internal wont have access to this. Mark this with internal or leave it out. Theres also a step missing which is any new functions that need to be made available need to be exported in `./index.ts`

**Review by rohit-kadhe — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (darshit-s3) [reply .github/pull_request_template.md]: done

File: .github/pull_request_template.md (hunk 3 above)
Comment (darshit-s3) [reply .github/pull_request_template.md:15]: This PR is just for vendor checklist. index.ts has non-vendor related methods.
docs is public, so it should be part of checklist

File: .github/pull_request_template.md (hunk 1 above)
Comment (darshit-s3) [reply .github/pull_request_template.md]: updated

**Review by rohit-kadhe — APPROVED** [review]

**Pull request merged**